internal/printer/print-to-tsx.go

```diff
@@ -134,15 +136,34 @@ func renderTsx(p *printer, n *Node) {
 			p.print("</Fragment>\n")
 		}
 		componentName := getTSXComponentName(p.opts.Filename)
+		propsIdent := props.Ident
+		paramsIdent := ""
+		if hasGetStaticPaths {
+			paramsIdent = "Get<InferredGetStaticPath, 'params'>"
+			if propsIdent == "Record<string, any>" {
+				propsIdent = "Get<InferredGetStaticPath, 'props'>"
+			}
+		}
 
-		p.print(fmt.Sprintf("export default function %s%s(_props: %s%s): any {}\n", componentName, props.Statement, props.Ident, props.Generics))
-		if props.Ident != "Record<string, any>" {
+		p.print(fmt.Sprintf("export default function %s%s(_props: %s%s): any {}\n", componentName, props.Statement, propsIdent, props.Generics))
+		if hasGetStaticPaths {
+			p.printf(`type ArrayElement<ArrayType extends readonly unknown[]> = ArrayType extends readonly (infer ElementType)[] ? ElementType : never;
```

Comment:
All those types should probably have some sort of marker on them (something like `ASTRO__Flattened`) because they end up in the global scope and could conflict with user code

Reply:
Yep, great catch! I've added `ASTRO__` as a prefix